File: documentation/user/en/solve/examples/filtering-products-in-category/product-listing.cs
EvitaResponse<ISealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		EntityLocaleEquals(CultureInfo.GetCultureInfo("en")),
        		HierarchyWithin(
        			"categories",
        			AttributeEquals("url", "/en/smartwatches")
        		),
        		AttributeEquals("status", "ACTIVE"),
        		Or(
        			AttributeInRangeNow("validity"),
        			AttributeIs("validity", Null)
        		),
        		ReferenceHaving(
        			"stocks",
        			AttributeGreaterThan("quantityOnStock", 0)
        		),
        		PriceInCurrency(new Currency("EUR")),
        		PriceInPriceLists("basic"),
        		PriceValidInNow()
        	),
        	Require(
        		EntityFetch(
        			AttributeContent("name"),
        			ReferenceContentWithAttributes(
        				"stocks",
        				AttributeContent("quantityOnStock")
        			),
        			PriceContentRespectingFilter("reference")
        		),
        		Page(1, 16)
        	)
        )
	)
);

File: documentation/user/en/solve/examples/filtering-products-in-category/product-listing.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					entityLocaleEquals(Locale.forLanguageTag("en")),
					hierarchyWithin(
						"categories",
						attributeEquals("url", "/en/smartwatches")
					),
					attributeEquals("status", "ACTIVE"),
					or(
						attributeInRangeNow("validity"),
						attributeIs("validity", NULL)
					),
					referenceHaving(
						"stocks",
						attributeGreaterThan("quantityOnStock", 0)
					),
					priceInCurrency(Currency.getInstance("EUR")),
					priceInPriceLists("basic"),
					priceValidInNow()
				),
				require(
					entityFetch(
						attributeContent("name"),
						referenceContentWithAttributes(
							"stocks",
							attributeContent("quantityOnStock")
						),
						priceContentRespectingFilter("reference")
					),
					page(1, 16)
				)
			)
		);
	}
);
